sim/common/cata/gurthalak.go

```diff
@@ -62,42 +53,36 @@ func init() {
 				},
 			})
 
-			procMask := character.GetDynamicProcMaskForWeaponEffect(gurthalakItemID)
-			aura := core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
-				Name:     "Gurthalak Trigger" + labelSuffix,
-				ActionID: core.ActionID{ItemID: gurthalakItemID},
-				Callback: core.CallbackOnSpellHitDealt,
-				ProcMask: core.ProcMaskMelee, // TODO: Verify correct proc masks on the PTR
-				Outcome:  core.OutcomeLanded,
-				Harmful:  true,
-				Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-					if _, ignore := ignoresSlot[spell.ActionID.SpellID]; !spell.ProcMask.Matches(*procMask) && !ignore {
-						return
-					}
-
-					if !sim.Proc(0.02, "Gurthalak, Voice of the Deeps"+labelSuffix) {
-						return
-					}
-
-					if sim.Log != nil {
-						slotLabel := core.Ternary(spell.IsMH(), "Main Hand", "Off Hand")
-						character.Log(sim, "Gurthalak (%s) procced by %s", slotLabel, spell.ActionID)
-					}
-
-					summonSpell.Cast(sim, result.Target)
-				},
-			})
-
-			character.ItemSwap.RegisterProc(gurthalakItemID, aura)
+			if character.Equipment.MainHand().ID == gurthalakItemID {
+				makeGurthalakProcTrigger(character, gurthalakItemID, summonSpell, labelSuffix, true)
+			}
+			if character.Equipment.OffHand().ID == gurthalakItemID {
+				makeGurthalakProcTrigger(character, gurthalakItemID, summonSpell, labelSuffix, false)
+			}
```

Comment:
This won't behave properly if Gurthalak is present in the swap set but not the main one, the Aura won't be registered at all in that scenario. I think you can fix it by doing
```
if (character.Equipment.MainHand().ID == gurthalakItemID) || (character.ItemSwap.GetUnequippedItemBySlot(proto.ItemSlot_ItemSlotMainHand).ID == gurthalakItemID) {
```
and likewise for the OH registration.

Reply:
I would probably add a function like we have for `CouldHaveSetBonus` called: `CouldHaveItem`